lib/vector-core/src/config/proxy.rs

```diff
@@ -105,32 +107,36 @@ impl ProxyConfig {
         }
     }
 
-    fn http_intercept(&self) -> Intercept {
-        self.interceptor().intercept("http")
-    }
-
-    fn http_proxy(&self) -> Result<Option<Proxy>, InvalidUri> {
-        self.http
+    fn build_proxy(
+        &self,
+        proxy_scheme: &'static str,
+        proxy_url: &Option<String>,
+    ) -> Result<Option<Proxy>, InvalidUri> {
+        proxy_url
             .as_ref()
             .map(|url| {
-                url.parse()
-                    .map(|parsed| Proxy::new(self.http_intercept(), parsed))
+                url.parse().map(|parsed| {
+                    let mut proxy = Proxy::new(self.interceptor().intercept(proxy_scheme), parsed);
+                    if let Ok(authority) = Url::parse(url) {
```

Comment:
Do we need to parse again? Don't we already have the parsed URL as `parsed`?

Reply:
AFAICT, parsing with `http::Uri` doesn't split out the username and password from the authority (see https://docs.rs/http/latest/http/uri/struct.Authority.html). Technically, putting basic auth in the authority is deprecated, so my guess is that they don't want to support it (even though it is pretty common still).

Reply:
I guess an alternative is that we parse with `url::Url` first, then convert its string representation to an `http::Uri`, but we still have to parse twice to populate both structs (or map all of the fields, but that is more verbose).

Reply:
related: https://github.com/hyperium/hyper/issues/1219

Reply:
Aha, yeah, I see, I didn't realize the original parse was into an `http::Uri` which doesn't have facilities for accessing the username/password part of the authority. 👍 to what you are doing here then.